clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/VeniceKafkaInputMapper.java

```diff
@@ -54,7 +59,7 @@ protected boolean process(
       BytesWritable keyBW,
       BytesWritable valueBW,
       Reporter reporter) {
-    if (veniceFilter.isPresent() && veniceFilter.get().applyRecursively(inputValue)) {
+    if (veniceFilterChain != null && !veniceFilterChain.isEmpty() && veniceFilterChain.apply(inputValue)) {
```

Comment:
`veniceFilterChain.isEmpty` check should be handled by `veniceFilterChain.apply`.